geopandas/tools/clip.py

```diff
@@ -168,7 +169,10 @@ def clip(gdf, mask, keep_geom_type=False):
     elif mask_is_list_like:
         box_mask = mask
     else:
-        box_mask = mask.bounds
+        # Avoid empty tuple returned by .bounds when geometry is empty. A tuple of
+        # all nan values is consitent with the behavior of
+        # {GeoSeries, GeoDataFrame}.total_bounds for empty geometries.
```

Comment:
```suggestion
        # {GeoSeries, GeoDataFrame}.total_bounds for empty geometries.
        # TODO(shapely) this can be removed once relying on Shapely 2.0
```

Shapely 2.0 will start to return a tuple of NaNs itself, so just adding a comment to remind us we can remove this special case once requiring shapely 2.0